march_joint_inertia_controller/include/march_joint_inertia_controller/inertia_estimator.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2020 Project March.
+
+#ifndef MARCH_WS_INERTIA_ESTIMATOR_H
+#define MARCH_WS_INERTIA_ESTIMATOR_H
+
+#include <hardware_interface/joint_command_interface.h>
+#include <ros/ros.h>
+#include <trajectory_interface/quintic_spline_segment.h>
+#include <urdf/model.h>
+
+/**
+ * \brief determines the absolute value of vector a and stores it into b so that a remains unchanged
+ *
+ * @param[in] a input vector with values that need to be converted to absolutes
+ * @param[out] b output vector with copies of the absolute values of a.
+ */
+void absolute(std::vector<double> a, std::vector<double>& b);
+
+/**
+ * \brief determiness the mean value of the given vector
+ *
+ * @param[in] a input vector from which the mean gets calculated
+ * @returns the mean value of a
+ */
+double mean(std::vector<double> a);
+
+/**
+ * \brief Class that bundles functionality to estimate inertia on a Revolute Joint.
+ *
+ * @param[in] joint the jointhandle used for obtaining current speed and effort.
+ * @param[in] lambda sets the lambda used in the RLS filter defaulted to 0.96
+ * @param[in] acc_size the size of the vector wherein the calculated acceleration is stored. This vector length is also
+ *  used in determining the length of the velocity vector and the filtered acceleration vector. It is of specific
+ *  size to calculate the alpha value. Defaulted to 12.
+ */
+class InertiaEstimator
+{
+public:
+  /**
+   * \brief Default constructor
+   */
+  InertiaEstimator(double lambda = 0.96, size_t acc_size = 12);
+
+  /**
+   * \brief Constructor that accepts a joint
+   */
+  InertiaEstimator(hardware_interface::JointHandle joint, double lambda = 0.96, size_t acc_size = 12);
+
+  /**
+   * \brief Destructor
+   */
+  ~InertiaEstimator();
+
+  /**
+   * \brief Get the position of the joint stored in the private variable
+   */
+  double getPosition();
+  double getAcceleration(unsigned int index);
+
+  void setJoint(hardware_interface::JointHandle joint);
+  void setLambda(double lambda);
+  void setAcc_size(size_t acc_size);
+  void setNodeHandle(ros::NodeHandle& nh);
+  /**
+   * \brief Initialize the publisher with a name
+   */
+  void configurePublisher(std::string name);
+  void publishInertia();
```

Comment:
I think it would be better to remove this publishing functionality from the inertia estimator and add it to the controller instead. Right now this requires all kind of extra methods to bloat this class, which clearly has a single function. Whereas the controller already has publishers and a node handle.

Reply:
Done!